wallet/src/main/java/com/android/identity_credential/wallet/logging/EventLogger.kt

```diff
@@ -67,10 +106,15 @@ class EventLogger(private val storageEngine: StorageEngine) {
             if (key.startsWith(STORAGE_KEY_PREFIX)) {
                 val value = storageEngine[key]
                 if (value != null) {
-                    when (val event = Event.fromCbor(value)) {
-                        is MdocPresentationEvent -> if (event.documentId == documentId) entries.add(event)
-                        is DocumentUpdateCheckEvent -> if (event.documentId == documentId) entries.add(event)
-                        // No need for an else branch here, as other Event subclasses don't have documentId
+                    try {
+                        val event = Event.fromCbor(value)
```

Comment:
The try/catch was added to make sure that you don't have a startup crashes that would happen if you already have events stored. Cbor deserializer would crash in that case because the stored objects have a different structure.

Reply:
Adding the try/catch is fine.

What I'm saying is: please restore the stuff inside the try/catch to the way it was: a when() statement rather than if() statements.